tools/scripts/upload-to-nr.js

```diff
@@ -33,137 +34,186 @@ var argv = yargs
+    var opts = {
+      uri: path,
+      method: 'GET',
+      gzip: true
+    }
 
-  asyncForEach(allFiles, readFile, cb)
+    console.log('downloading ', path)
 
-  function readFile (file, next) {
-    fs.readFile(path.resolve(__dirname, '../../build/', file), function (err, data) {
-      if (err) return next(err)
-      fileData[file] = data
-      next()
+    return new Promise((resolve, reject) => {
+      request(opts, (err, res, body) => {
+        if (err || res.statusCode !== 200) {
+          reject(err)
+          return
+        }
+        resolve([path, fileName, body])
+      })
     })
   }
-}
 
-function uploadAllLoadersToDB (environment, cb) {
-  asyncForEach(loaders, function (filename, next) {
-    uploadLoaderToDB(filename, fileData[filename], environment, next)
-  }, cb, uploadErrorCallback)
-}
-
-function uploadLoaderToDB (filename, loader, environment, cb) {
-  var baseOptions = {
-    method: 'PUT',
-    followAllRedirects: true,
-    json: {
-      js_agent_loader: {
-        version: filename, // sic
-        loader: loader.toString()
-      },
-      set_current: false
+  /**
+   * Upload a loader to NRDB
+   * @param {string} filename 
+   * @param {string} loader 
+   * @param {string} environment 
+   * @param {Function} cb 
+   */
+  function uploadLoaderToDB(filename, loader, environment, cb) {
+    var baseOptions = {
+      method: 'PUT',
+      followAllRedirects: true,
+      json: {
+        js_agent_loader: {
+          version: filename, // sic
+          loader
+        },
+        set_current: false
+      }
     }
-  }
 
-  var envOptions = {
-    staging: {
-      url: 'https://staging-api.newrelic.com/v2/js_agent_loaders/create.json',
-      headers: {
-        'X-Api-Key': argv['staging-api-key']
-      }
-    },
-    eu: {
-      url: 'https://api.eu.newrelic.com/v2/js_agent_loaders/create.json',
-      headers: {
-        'X-Api-Key': argv['eu-api-key']
-      }
-    },
-    production: {
-      url: 'https://api.newrelic.com/v2/js_agent_loaders/create.json',
-      headers: {
-        'X-Api-Key': argv['production-api-key']
+    var envOptions = {
+      staging: {
+        url: 'https://staging-api.newrelic.com/v2/js_agent_loaders/create.json',
+        headers: {
+          'X-Api-Key': argv['staging-api-key']
+        }
+      },
+      eu: {
+        url: 'https://api.eu.newrelic.com/v2/js_agent_loaders/create.json',
+        headers: {
+          'X-Api-Key': argv['eu-api-key']
+        }
+      },
+      production: {
+        url: 'https://api.newrelic.com/v2/js_agent_loaders/create.json',
+        headers: {
+          'X-Api-Key': argv['production-api-key']
+        }
       }
     }
-  }
 
-  console.log('Uploading loader ' + filename + ' to ' + environment + '...')
-  var options = util._extend(envOptions[environment], baseOptions)
+    console.log('Uploading loader ' + filename + ' to ' + environment + '...')
+    var options = util._extend(envOptions[environment], baseOptions)
```

Comment:
This could be replace with object spread syntax and we can remove reliance on this third-party.

```suggestion
    var options = {
      ...envOptions[environment],
      ...baseOptions
    }
```